packages/fluentui/react-bindings/src/hooks/useClasses.ts

```diff
@@ -0,0 +1,218 @@
+/* eslint-disable */
+
+import hash from '@emotion/hash';
+import { RendererParam } from '@fluentui/react-northstar-styles-renderer';
+import {
+  ComponentSlotStyle,
+  ComponentSlotStylesInput,
+  ComponentSlotStylesPrepared,
+  ComponentStyleFunctionParam,
+  ComponentVariablesInput,
+  ComponentVariablesObject,
+  ICSSInJSStyle,
+  mergeComponentStyles,
+  withDebugId,
+} from '@fluentui/styles';
+import cx from 'classnames';
+import * as _ from 'lodash';
+
+import { useFluentContext } from '../context';
+import { ComponentDesignProp, PrimitiveProps } from '../styles/types';
+
+type UseStylesOptions<StyleProps extends PrimitiveProps> = {
+  /** A classname that will be added by default to all instances of a component on the `root` slot. */
+  className?: string;
+
+  /**
+   * A mapping from component's props to styles functions props. Can be only primitive types as they will be used for
+   * cache keys.
+   */
+  mapPropsToStyles?: () => StyleProps;
+
+  /**
+   * A set props of that contain mapping for props that perform inline styles overrides, for example `styles` or
+   * `variables`.
+   */
+  mapPropsToInlineStyles?: () => InlineStyleProps<StyleProps>;
+};
+
+type InlineStyleProps<StyleProps> = {
+  /** Additional CSS class name(s) to apply.  */
+  className?: string;
+
+  design?: ComponentDesignProp;
+
+  /** Additional CSS styles to apply to the component instance.  */
+  styles?: ComponentSlotStyle<StyleProps, any>; // TODO: see if we can improve it
+
+  /** Override for theme site variables to allow modifications of component styling via themes. */
+  variables?: ComponentVariablesInput;
+};
+
+export const useClasses = <StyleProps extends PrimitiveProps>(
+  displayName: string,
+  options: UseStylesOptions<StyleProps>,
+): string => {
+  const { disableAnimations, performance: performanceFlags, telemetry, theme, renderer, rtl } = useFluentContext();
+  const {
+    className: componentClassName = process.env.NODE_ENV === 'production' ? '' : 'no-classname-🙉',
+    mapPropsToStyles = () => ({} as StyleProps),
+    mapPropsToInlineStyles = () => ({} as InlineStyleProps<StyleProps>),
+  } = options;
+
+  const componentStylesProps = mapPropsToStyles();
+  const inlineStylesProps = mapPropsToInlineStyles();
+
+  //
+  // To compute styles we are going through three stages:
+  // - resolve variables (siteVariables => componentVariables + props.variables)
+  // - resolve styles (with resolvedVariables & props.styles & props.design)
+  // - compute classes (with resolvedStyles)
+  // - conditionally add sources for evaluating debug information to component
+
+  const telemetryVariablesStart = telemetry?.enabled ? performance.now() : 0;
+
+  let resolvedVariablesHash: string = '';
+  let resolvedVariables: ComponentVariablesObject = {};
```

Comment:
maybe you can create a helper function to return these two variables with the logic below that sets them ?

Reply:
i.e. `resolveComponentVariables(theme.componentVariables[displayName])`